### ISSUE TITLE
Unexpected results

### ISSUE THREAD

**Issue by MikepdXRider** [brian3kb/graham_scan_js#16]
### Input Data In LatLng - Actual data is google maps markers. 
```
[
    {
        "lng": -89.04826539999999,
        "lat": 40.4572814
    },
    {
        "lng": -89.04826539999999,
        "lat": 40.4572814
    },
    {
        "lng": -89.04632170000001,
        "lat": 40.4528266
    },
    {
        "lng": -89.0462646,
        "lat": 40.4524364
    },
    {
        "lng": -89.0462652,
        "lat": 40.4521928
    },
    {
        "lng": -89.0461206,
        "lat": 40.45196019999999
    },
    {
        "lng": -89.0460007,
        "lat": 40.4516839
    },
    {
        "lng": -89.045872,
        "lat": 40.4514006
    },
    {
        "lng": -89.0460978,
        "lat": 40.4530237
    },
    {
        "lng": -89.04557539999999,
        "lat": 40.4529
    },
    {
        "lng": -89.045594,
        "lat": 40.4526814
    },
    {
        "lng": -89.04566489999999,
        "lat": 40.4523262
    },
    {
        "lng": -89.04552729999999,
        "lat": 40.4520635
    },
    {
        "lng": -89.0454006,
        "lat": 40.4518228
    },
    {
        "lng": -89.04035069999999,
        "lat": 40.4511462
    },
    {
        "lng": -89.03999619999999,
        "lat": 40.4512826
    },
    {
        "lng": -89.04089189999999,
        "lat": 40.4515287
    },
    {
        "lng": -89.0401569,
        "lat": 40.4516308
    },
    {
        "lng": -89.0410225,
        "lat": 40.4512711
    },
    {
        "lng": -89.0440267,
        "lat": 40.4530663
    },
    {
        "lng": -89.0434483,
        "lat": 40.4528427
    },
    {
        "lng": -89.04296169999999,
        "lat": 40.45276339999999
    },
    {
        "lng": -89.04245619999999,
        "lat": 40.4526677
    },
    {
        "lng": -89.0419369,
        "lat": 40.4525717
    },
    {
        "lng": -89.0410412,
        "lat": 40.452246
    },
    {
        "lng": -89.04081719999999,
        "lat": 40.4520054
    },
    {
        "lng": -89.0440267,
        "lat": 40.4533845
    },
    {
        "lng": -89.0426086,
        "lat": 40.4532137
    },
    {
        "lng": -89.0422354,
        "lat": 40.4531311
    },
    {
        "lng": -89.04178759999999,
        "lat": 40.4530478
    },
    {
        "lng": -89.0413397,
        "lat": 40.45296460000001
    },
    {
        "lng": -89.0409852,
        "lat": 40.452902
    },
    {
        "lng": -89.0406493,
        "lat": 40.45274
    },
    {
        "lng": -89.04029469999999,
        "lat": 40.4524785
    },
    {
        "lng": -89.0401455,
        "lat": 40.4522386
    },
    {
        "lng": -89.0452517,
        "lat": 40.4514698
    },
    {
        "lng": -89.04486209999999,
        "lat": 40.4510384
    },
    {
        "lng": -89.04441849999999,
        "lat": 40.4511798
    },
    {
        "lng": -89.044176,
        "lat": 40.4510784
    },
    {
        "lng": -89.0440423,
        "lat": 40.4506969
    },
    {
        "lng": -89.04353669999999,
        "lat": 40.4507898
    },
    {
        "lng": -89.0432398,
        "lat": 40.4506543
    },
    {
        "lng": -89.04353669999999,
        "lat": 40.4507898
    },
    {
        "lng": -89.0435677,
        "lat": 40.4504348
    },
    {
        "lng": -89.04395799999999,
        "lat": 40.4502574
    },
    {
        "lng": -89.045814,
        "lat": 40.4511397
    },
    {
        "lng": -89.04547749999999,
        "lat": 40.4508854
    },
    {
        "lng": -89.0456906,
        "lat": 40.4505941
    },
    {
        "lng": -89.0456572,
        "lat": 40.4503141
    },
    {
        "lng": -89.0456143,
        "lat": 40.4500338
    },
    {
        "lng": -89.0451496,
        "lat": 40.4508172
    },
    {
        "lng": -89.0451136,
        "lat": 40.4505815
    },
    {
        "lng": -89.0450795,
        "lat": 40.4503586
    },
    {
        "lng": -89.0442044,
        "lat": 40.4504944
    },
    {
        "lng": -89.044265,
        "lat": 40.4503412
    },
    {
        "lng": -89.0481501,
        "lat": 40.4509718
    },
    {
        "lng": -89.04124639999999,
        "lat": 40.4504773
    },
    {
        "lng": -89.041433,
        "lat": 40.45024009999999
    },
    {
        "lng": -89.0416383,
        "lat": 40.450023
    },
    {
        "lng": -89.0417129,
        "lat": 40.4497054
    },
    {
        "lng": -89.04178759999999,
        "lat": 40.44946729999999
    },
    {
        "lng": -89.0418622,
        "lat": 40.4491496
    },
    {
        "lng": -89.0409852,
        "lat": 40.4507536
    },
    {
        "lng": -89.0404476,
        "lat": 40.4502308
    },
    {
        "lng": -89.04081719999999,
        "lat": 40.4500162
    },
    {
        "lng": -89.04094789999999,
        "lat": 40.4497587
    },
    {
        "lng": -89.0410971,
        "lat": 40.4495212
    },
    {
        "lng": -89.0411158,
        "lat": 40.449223
    },
    {
        "lng": -89.0411904,
        "lat": 40.4489849
    },
    {
        "lng": -89.0411904,
        "lat": 40.4486666
    },
    {
        "lng": -89.0392497,
        "lat": 40.4488893
    },
    {
        "lng": -89.039399,
        "lat": 40.4485722
    },
    {
        "lng": -89.0393244,
        "lat": 40.4483329
    },
    {
        "lng": -89.0394737,
        "lat": 40.4480159
    },
    {
        "lng": -89.0395297,
        "lat": 40.4477577
    },
    {
        "lng": -89.0391751,
        "lat": 40.4494457
    },
    {
        "lng": -89.0392262,
        "lat": 40.4491256
    },
    {
        "lng": -89.0392497,
        "lat": 40.4488893
    },
    {
        "lng": -89.039399,
        "lat": 40.4485722
    },
    {
        "lng": -89.0393244,
        "lat": 40.4483329
    },
    {
        "lng": -89.0394737,
        "lat": 40.4480159
    },
    {
        "lng": -89.0395297,
        "lat": 40.4474792
    },
    {
        "lng": -89.03857789999999,
        "lat": 40.4496794
    },
    {
        "lng": -89.0391751,
        "lat": 40.4494457
    }
]
```

### Hull results
```
[
    {
        "x": -89.04826539999999,
        "y": 40.4572814
    },
    {
        "x": -89.0481501,
        "y": 40.4509718
    },
    {
        "x": -89.0456143,
        "y": 40.4500338
    },
    {
        "x": -89.0456906,
        "y": 40.4505941
    },
    {
        "x": -89.045872,
        "y": 40.4514006
    },
    {
        "x": -89.04632170000001,
        "y": 40.4528266
    },
    {
        "x": -89.0450795,
        "y": 40.4503586
    },
    {
        "x": -89.0451496,
        "y": 40.4508172
    },
    {
        "x": -89.0452517,
        "y": 40.4514698
    },
    {
        "x": -89.045594,
        "y": 40.4526814
    },
    {
        "x": -89.0442044,
        "y": 40.4504944
    },
    {
        "x": -89.0440423,
        "y": 40.4506969
    },
    {
        "x": -89.044176,
        "y": 40.4510784
    },
    {
        "x": -89.0435677,
        "y": 40.4504348
    },
    {
        "x": -89.0432398,
        "y": 40.4506543
    },
    {
        "x": -89.0418622,
        "y": 40.4491496
    },
    {
        "x": -89.0417129,
        "y": 40.4497054
    },
    {
        "x": -89.0416383,
        "y": 40.450023
    },
    {
        "x": -89.0395297,
        "y": 40.4477577
    },
    {
        "x": -89.0393244,
        "y": 40.4483329
    },
    {
        "x": -89.0392497,
        "y": 40.4488893
    },
    {
        "x": -89.0434483,
        "y": 40.4528427
    },
    {
        "x": -89.0440267,
        "y": 40.4533845
    },
    {
        "x": -89.0392262,
        "y": 40.4491256
    },
    {
        "x": -89.0391751,
        "y": 40.4494457
    },
    {
        "x": -89.0410225,
        "y": 40.4512711
    },
    {
        "x": -89.04245619999999,
        "y": 40.4526677
    },
    {
        "x": -89.03857789999999,
        "y": 40.4496794
    },
    {
        "x": -89.0401455,
        "y": 40.4522386
    },
    {
        "x": -89.04029469999999,
        "y": 40.4524785
    },
    {
        "x": -89.0406493,
        "y": 40.45274
    }
]
```

### Visual Results
<img width="123" alt="image" src="https://github.com/brian3kb/graham_scan_js/assets/79485147/3b718442-2331-46cc-821c-8a10c3e5d7ce">

### Implementation
```js
function ConvexHull(markers) {
    // Functions
    this.pointArrayToLatLng = function(points) {
        var output = [];
        for (var i = 0; i < points.length; i++) {
            output.push(new google.maps.LatLng(points[i].y, points[i].x));
        }
        return output;
    }

    this.getConvexHull = function() {
        return hull;
    }

    // Convenience wrapper
    this.setMap = function(map) {
        hull.setMap(map);
    }

    // Set up
    var convexHull = new ConvexHullGrahamScan();
    for (var i = 0; i < markers.length; i++) {
        convexHull.addPoint(markers[i].position.lng(), markers[i].position.lat());
    }
    var hullPoints = convexHull.getHull();

    // Create polygon
    var hull = new google.maps.Polygon({
        paths: this.pointArrayToLatLng(hullPoints),
        strokeColor: '#4285F4',
        strokeWeight: 1,
        fillColor: '#4285F4',
        fillOpacity: 0.4
    });
}
```

### What I've tried
- Remove duplicates from input data. 
- Convert points to strings per #11; seemed to improve the issue in some areas, but make it worst in others. The following image is the same input data using strings.
<img width="107" alt="image" src="https://github.com/brian3kb/graham_scan_js/assets/79485147/110f99a5-36f2-4783-a3af-da779d22d7e2">

Here's another example of the string behavior. The first image is using numbers, and the second uses strings. 

![image](https://github.com/brian3kb/graham_scan_js/assets/79485147/0ea54629-567e-41c9-8b35-8c59c4d5229e)

![image](https://github.com/brian3kb/graham_scan_js/assets/79485147/1ff1eb33-2fa8-4adb-ac01-df7c3e2f82a0)

### Notes
I have more extreme examples of unexpected results. The provided example contains a relatively smaller set of data and I wanted to keep the issue simple. Here are some other results I'm seeing. Let me know if having input data for any of these would also be helpful. 
<img width="101" alt="image" src="https://github.com/brian3kb/graham_scan_js/assets/79485147/49f8eb2c-f586-4f37-9421-d789f95a80cd">
<img width="80" alt="image" src="https://github.com/brian3kb/graham_scan_js/assets/79485147/99a4c070-f8ad-4db5-b8a5-e43c8d5f0c06">
<img width="87" alt="image" src="https://github.com/brian3kb/graham_scan_js/assets/79485147/c233c767-315c-495b-b66d-aca3aafd5d33">
<img width="97" alt="image" src="https://github.com/brian3kb/graham_scan_js/assets/79485147/7979c9a1-b1d8-4a1a-8bd3-0d702678603d">

It feels like ~50% of the results I'm getting are like this.

**Comment by brian3kb** [comment]
With your given coords array, I'm not seeing the same result you are, I got a consistent 7 point hull.

```javascript
function initialize() {

    const coords = [
        {
            "lng": -89.04826539999999,
            "lat": 40.4572814
        },
        {
            "lng": -89.04826539999999,
            "lat": 40.4572814
        },
        {
            "lng": -89.04632170000001,
            "lat": 40.4528266
        },
        {
            "lng": -89.0462646,
            "lat": 40.4524364
        },
        {
            "lng": -89.0462652,
            "lat": 40.4521928
        },
        {
            "lng": -89.0461206,
            "lat": 40.45196019999999
        },
        {
            "lng": -89.0460007,
            "lat": 40.4516839
        },
        {
            "lng": -89.045872,
            "lat": 40.4514006
        },
        {
            "lng": -89.0460978,
            "lat": 40.4530237
        },
        {
            "lng": -89.04557539999999,
            "lat": 40.4529
        },
        {
            "lng": -89.045594,
            "lat": 40.4526814
        },
        {
            "lng": -89.04566489999999,
            "lat": 40.4523262
        },
        {
            "lng": -89.04552729999999,
            "lat": 40.4520635
        },
        {
            "lng": -89.0454006,
            "lat": 40.4518228
        },
        {
            "lng": -89.04035069999999,
            "lat": 40.4511462
        },
        {
            "lng": -89.03999619999999,
            "lat": 40.4512826
        },
        {
            "lng": -89.04089189999999,
            "lat": 40.4515287
        },
        {
            "lng": -89.0401569,
            "lat": 40.4516308
        },
        {
            "lng": -89.0410225,
            "lat": 40.4512711
        },
        {
            "lng": -89.0440267,
            "lat": 40.4530663
        },
        {
            "lng": -89.0434483,
            "lat": 40.4528427
        },
        {
            "lng": -89.04296169999999,
            "lat": 40.45276339999999
        },
        {
            "lng": -89.04245619999999,
            "lat": 40.4526677
        },
        {
            "lng": -89.0419369,
            "lat": 40.4525717
        },
        {
            "lng": -89.0410412,
            "lat": 40.452246
        },
        {
            "lng": -89.04081719999999,
            "lat": 40.4520054
        },
        {
            "lng": -89.0440267,
            "lat": 40.4533845
        },
        {
            "lng": -89.0426086,
            "lat": 40.4532137
        },
        {
            "lng": -89.0422354,
            "lat": 40.4531311
        },
        {
            "lng": -89.04178759999999,
            "lat": 40.4530478
        },
        {
            "lng": -89.0413397,
            "lat": 40.45296460000001
        },
        {
            "lng": -89.0409852,
            "lat": 40.452902
        },
        {
            "lng": -89.0406493,
            "lat": 40.45274
        },
        {
            "lng": -89.04029469999999,
            "lat": 40.4524785
        },
        {
            "lng": -89.0401455,
            "lat": 40.4522386
        },
        {
            "lng": -89.0452517,
            "lat": 40.4514698
        },
        {
            "lng": -89.04486209999999,
            "lat": 40.4510384
        },
        {
            "lng": -89.04441849999999,
            "lat": 40.4511798
        },
        {
            "lng": -89.044176,
            "lat": 40.4510784
        },
        {
            "lng": -89.0440423,
            "lat": 40.4506969
        },
        {
            "lng": -89.04353669999999,
            "lat": 40.4507898
        },
        {
            "lng": -89.0432398,
            "lat": 40.4506543
        },
        {
            "lng": -89.04353669999999,
            "lat": 40.4507898
        },
        {
            "lng": -89.0435677,
            "lat": 40.4504348
        },
        {
            "lng": -89.04395799999999,
            "lat": 40.4502574
        },
        {
            "lng": -89.045814,
            "lat": 40.4511397
        },
        {
            "lng": -89.04547749999999,
            "lat": 40.4508854
        },
        {
            "lng": -89.0456906,
            "lat": 40.4505941
        },
        {
            "lng": -89.0456572,
            "lat": 40.4503141
        },
        {
            "lng": -89.0456143,
            "lat": 40.4500338
        },
        {
            "lng": -89.0451496,
            "lat": 40.4508172
        },
        {
            "lng": -89.0451136,
            "lat": 40.4505815
        },
        {
            "lng": -89.0450795,
            "lat": 40.4503586
        },
        {
            "lng": -89.0442044,
            "lat": 40.4504944
        },
        {
            "lng": -89.044265,
            "lat": 40.4503412
        },
        {
            "lng": -89.0481501,
            "lat": 40.4509718
        },
        {
            "lng": -89.04124639999999,
            "lat": 40.4504773
        },
        {
            "lng": -89.041433,
            "lat": 40.45024009999999
        },
        {
            "lng": -89.0416383,
            "lat": 40.450023
        },
        {
            "lng": -89.0417129,
            "lat": 40.4497054
        },
        {
            "lng": -89.04178759999999,
            "lat": 40.44946729999999
        },
        {
            "lng": -89.0418622,
            "lat": 40.4491496
        },
        {
            "lng": -89.0409852,
            "lat": 40.4507536
        },
        {
            "lng": -89.0404476,
            "lat": 40.4502308
        },
        {
            "lng": -89.04081719999999,
            "lat": 40.4500162
        },
        {
            "lng": -89.04094789999999,
            "lat": 40.4497587
        },
        {
            "lng": -89.0410971,
            "lat": 40.4495212
        },
        {
            "lng": -89.0411158,
            "lat": 40.449223
        },
        {
            "lng": -89.0411904,
            "lat": 40.4489849
        },
        {
            "lng": -89.0411904,
            "lat": 40.4486666
        },
        {
            "lng": -89.0392497,
            "lat": 40.4488893
        },
        {
            "lng": -89.039399,
            "lat": 40.4485722
        },
        {
            "lng": -89.0393244,
            "lat": 40.4483329
        },
        {
            "lng": -89.0394737,
            "lat": 40.4480159
        },
        {
            "lng": -89.0395297,
            "lat": 40.4477577
        },
        {
            "lng": -89.0391751,
            "lat": 40.4494457
        },
        {
            "lng": -89.0392262,
            "lat": 40.4491256
        },
        {
            "lng": -89.0392497,
            "lat": 40.4488893
        },
        {
            "lng": -89.039399,
            "lat": 40.4485722
        },
        {
            "lng": -89.0393244,
            "lat": 40.4483329
        },
        {
            "lng": -89.0394737,
            "lat": 40.4480159
        },
        {
            "lng": -89.0395297,
            "lat": 40.4474792
        },
        {
            "lng": -89.03857789999999,
            "lat": 40.4496794
        },
        {
            "lng": -89.0391751,
            "lat": 40.4494457
        }
    ];

    const centrePoint = new google.maps.LatLng(40.453, -89.04);

    const mapOptions = {
        zoom: 16,
        center: centrePoint,
        mapTypeId: google.maps.MapTypeId.ROADMAP
    };

    const map = new google.maps.Map(document.getElementById('map-canvas'),
        mapOptions);

    const hull = ConvexHull(coords.map(item => new google.maps.Marker({
        position: new google.maps.LatLng(item.lat, item.lng),
        map: map
    })));

    hull.setMap(map);
}

function ConvexHull(markers) {
    // Functions
    this.pointArrayToLatLng = function(points) {
        var output = [];
        for (var i = 0; i < points.length; i++) {
            output.push(new google.maps.LatLng(points[i].y, points[i].x));
        }
        return output;
    }

    this.getConvexHull = function() {
        return hull;
    }

    // Convenience wrapper
    this.setMap = function(map) {
        hull.setMap(map);
    }

    // Set up
    var convexHull = new ConvexHullGrahamScan();
    for (var i = 0; i < markers.length; i++) {
        convexHull.addPoint(markers[i].position.lng(), markers[i].position.lat());
    }
    var hullPoints = convexHull.getHull();

    console.warn(this.pointArrayToLatLng(hullPoints));

    // Create polygon
    var hull = new google.maps.Polygon({
        paths: this.pointArrayToLatLng(hullPoints),
        strokeColor: '#4285F4',
        strokeWeight: 1,
        fillColor: '#4285F4',
        fillOpacity: 0.4
    });
    return hull;
}

google.maps.event.addDomListener(window, 'load', initialize);
``` 

Giving a convex hull drawn to the map

![image](https://github.com/brian3kb/graham_scan_js/assets/1716071/709ab889-01d6-4185-8289-2806f10e503f)

I see 7 points for this particular example data-set, how many points are you getting back? It should be the same number, and if that is the case, then its is something on the polygon mapping side possibly joining the points in a different order?

**Comment by brian3kb** [comment]
Looking at the 1.0.5 release, the minified version was at 1.0.3 not 1.0.5 - not sure if that is the issue, but try grabbing the latest release and see if that fixes things, as your points array and the output hull looks good. I also added an example using the points array you provided in this issue https://brian3kb.github.io/graham_scan_js/pages/gmaps_example6.html

**Comment by MikepdXRider** [comment]
Thanks for the detailed and informative response @brian3kb! I really appreciate your effort to reproduce the issue and provide a working example with the provided data. 

> I see 7 points for this particular example data-set, how many points are you getting back?

I believe I got 31 points for this particular data set. 

I noticed the [npm package](https://www.npmjs.com/package/graham_scan) has 1.0.4 listed as the latest release version and the minified version appears to be 1.0.4. I also see that 1.0.5 is referenced in this repo. I can install the package via GitHub and test but am curious if it's your intention to publish v1.0.5 to the npm registry. 

Edit: Following up to share that updating from 1.0.4 to 1.0.5 appears to have resolved this issue. Still curious about the intention to publish the latest version of the package to npm.
